compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -3490,4 +3492,35 @@ mod control_dependence {
         let loop_ = loops.yet_to_unroll.pop().unwrap();
         assert!(!loop_.is_fully_executed(&loops.cfg));
     }
+
+    #[test_case(RuntimeType::Brillig(InlineType::default()))]
+    #[test_case(RuntimeType::Acir(InlineType::default()))]
+    fn do_not_hoist_unsafe_array_get_from_control_dependent_block(runtime: RuntimeType) {
+        // We use an unknown index v0 to index an array, but only if v1 is true,
+        // so we should not hoist the constraint or the array_get into the header.
```

Comment:
Maybe also just add that in Brillig array get operations will not fail when OOB, but they could return incorrect data, causing knockdown loop invariant instructions to fail when the loop is not meant to fail.